osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/security/ReplaceKeyCommandExecutor.java

```diff
@@ -130,17 +130,28 @@ public DlmsDevice execute(
       final MessageMetadata messageMetadata)
       throws OsgpException {
 
+    ReplaceKeyInput keyWrapper2 = keyWrapper;
```

Comment:
Why do you create keyWrapper2 here? The assignment with keyWrapper is not used. On line 144 another value is assigned.

Reply:
The assignment of another value of keyWrapper is done in a if statement: _if(!keyWrapper.isGenerated())_.
The initial assignment to keyWrapper2 might be more clean if done in an else statement